lib/transport/transport-aux-data.h

```diff
@@ -81,6 +81,16 @@ log_transport_aux_data_copy(LogTransportAuxData *dst, LogTransportAuxData *src)
     }
 }
 
+static inline void
+log_transport_aux_data_move(LogTransportAuxData *dst, LogTransportAuxData *src)
+{
+  gsize data_to_copy = sizeof(*src) - sizeof(src->data) + src->end_ptr;
+
+  if (dst)
+    memcpy(dst, src, data_to_copy);
```

Comment:
This feels weird here as `LogTransportAuxData` is owned by the `LogTransportStack`. Am I missing something here?

Reply:
This function moves the responsibility from one LogTransportAuxData instance to another, e.g. it copies all
data & references while dropping it from the old one.

It is similar to log_transport_aux_copy() but also drops references from the old one. This is called when we are moving all of our state from one LogTransportStack to another. This happens when we replace the LogProto implementation on the fly.

Any why we move, instead of just having a reference, that's easy too: I did not want to add an extra dereference a pointer to LogTransportStack every time we need to do a read or write in a LogProto instance.

Reply:
I hope this clears this up. I am resolving this conversation now, please reopen if you disagree.